Code/Editor/Platform/Mac/editor_mac.cmake

```diff
@@ -13,3 +13,27 @@ set_target_properties(Editor PROPERTIES
     RESOURCE ${CMAKE_CURRENT_LIST_DIR}/Images.xcassets
     XCODE_ATTRIBUTE_ASSETCATALOG_COMPILER_APPICON_NAME EditorAppIcon
 )
+
+# We cannot use ly_add_target here because we're already including this file from inside ly_add_target
+# So we need to setup target, dependencies and install logic manually.
+add_executable(EditorDummy Platform/Mac/main_dummy.cpp)
+add_executable(AZ::EditorDummy ALIAS EditorDummy)
+
+ly_target_link_libraries(EditorDummy
+    PRIVATE
+        AZ::AzCore
+        AZ::AzFramework)
+
+ly_add_dependencies(Editor EditorDummy)
+
+# Store the aliased target into a DIRECTORY property
+set_property(DIRECTORY APPEND PROPERTY LY_DIRECTORY_TARGETS AZ::EditorDummy)
+
+# Store the directory path in a GLOBAL property so that it can be accessed
+# in the layout install logic. Skip if the directory has already been added
+get_property(ly_all_target_directories GLOBAL PROPERTY LY_ALL_TARGET_DIRECTORIES)
+if(NOT CMAKE_CURRENT_SOURCE_DIR IN_LIST ly_all_target_directories)
+    set_property(GLOBAL APPEND PROPERTY LY_ALL_TARGET_DIRECTORIES ${CMAKE_CURRENT_SOURCE_DIR})
+endif()
```

Comment:
I can create a function to wrap this and put it in LYWrapper_mac.cmake so that I can just call it from the AssetProcessor's mac cmake as well. I decided against it in the end because I felt that doing it in this file makes it more clear why we can't just use ly_add_target() and this is also sort of a niche requirement that really doesn't need to be used in a lot of places. But having said that, I'm open to suggestions on better ways to handle this while also not having to spill over to other platforms.

Reply:
yes, I would just create another target that is generic, e.g. "ApplicationRunner" and make a `ly` cmake function that can basically "instantiate` cases of that binary passing some defines that will build it for different cases (e.g. Editor/AP/etc)
Because we will have more applications trying to run and they all need this trick

(Can be done in a different PR)

Reply:
+1 to making a configurable "ApplicationRunner" target for this and AP to use.